moengage-segment-kotlin-destination/src/main/java/com/segment/analytics/kotlin/destinations/moengage/MoEngageDestination.kt

```diff
@@ -126,6 +126,14 @@ class MoEngageDestination(private val application: Application) : DestinationPlu
         try {
             super.identify(payload)
             Logger.print { "$tag identify(): will try to track $payload" }
+            if (payload.userId.isNotEmpty()) {
```

Comment:
I don't think this check is required as once the user id is set by the client same userId will be passed every time even if the client doesn't set the userId next time

If the user is setting the empty or blank userId, it is already filtered in the Native SDK

Reply:
Okay